tools/python/boutiques/conftest.py

```diff
@@ -0,0 +1,43 @@
+import json, tempfile
+
+def pytest_addoption(parser):
+    parser.addoption("--descriptor", action="append", default=[])
+
+
+def fetch_tests(descriptor_filename):
+    
+    descriptor = json.loads(open(descriptor_filename).read())
+    tests = [];    
+
+    # For each test present in the descriptor:
+    for test in descriptor["tests"]:
+
+        # We first extract the invocation and put it inside a temporary file.
+        invocation_JSON = json.dumps(test["invocation"])
+        temp_invocation_JSON = tempfile.NamedTemporaryFile(suffix=".json", 
+                                                           delete=False)
+        temp_invocation_JSON.write(invocation_JSON.encode())
+        temp_invocation_JSON.seek(0)
+        
+        # Now we setup the necessary elements for the testing function.
+        tests.append([descriptor_filename, test, temp_invocation_JSON])
+        
+    return (descriptor["name"], tests)
+
+# This function will be executed by pytest before proceeding to the actual testing
+def pytest_generate_tests(metafunc):
+    descriptor_filename =  metafunc.config.getoption('descriptor')
+
+    # Each element in 'tests' will hold the necessary informations for a single test
+    # Those informations are:
+    #                         . The descriptor (common to all)
+    #                         . The related JSON data, describing the test 
+    #                         (more convenient, no need to extract again from descriptor)
+    #                         . The invocation file needed for the test
+    descriptor_name, tests = fetch_tests(descriptor_filename[0])
+
+    # Generate the test ids for each of the test cases.
+    # An id is created by concatenaning the name of the descriptor with the name of the test case.
+    names = ['\'' + descriptor_name + "\'-\'" + parameters[1]["name"] + '\'' for parameters in tests]
```

Comment:
For readability, I'd write this as: 
```
"{}-{}".format(descriptor_name, params[1]["name"] for params in tests]
```

Out of curiosity, how come you added extra `'`s?

Reply:
I added extra ' because the name of the descriptor can take whitespaces and any other characters. If I run one of the JSON file which include failing test cases, pytest shows me "Example Boutique Tool-test1" (or something just like that). I thought including ' would better delimit the name of the descriptor and the test case.

I actually wasn't so confident with this idea. Feel free to share a suggestion on that.

Reply:
OK, so how about you replace spaces with `-` and separate descriptor name with a `_`? I.e:

```
"{}_{}".format(descriptor_name, params[1]["name"].replace(' ', '-') for params in tests]
```

Reply:
Are you sure you want to modify the name of the descriptor ? I personally think it's better not to alter it.
However what could be done is that we could modify the schema so as to disallow white-spaces and `_`, but that might be overkill for such a small concern.

Another idea I had was to format it like that: `desc:"Exemple Test Tool"; test:"test1"`. But I thought not to go that way because it doesn't really feel much like an ID anymore (which might have appliances other than just displaying tests that failed).

Reply:
@Trajan64 I'm not sure I understand... With what you proposed, a descriptor named "`mydesc.json`", and two tests, "`my test 1`" and "`my test 2`", we'd get:
```
'mydesc.json'-'my test 1'
'mydesc.json'-'my test 2'
```

With my proposal, we'd get:
```
mydesc.json_my-test-1
mydesc.json_my-test-2
```

I prefer mine simply because it has no spaces or special characters in it. What's the issue?

Reply:
Oh okay, I think we have a confusion on the matter of the descriptor name.
For me the descriptor name is the name object inside the descriptor, and not the filename.
For `tests_good.json`, there is the following property:  `"name": "Example Boutiques Tool"`
So as a consequence, with what I originally proposed and `tests_good.json` we have `'Exemple Test Tool'-'test1'` for one of the tests.

And also, I read your code too fast. I was under the impression that you were altering the white spaces within the descriptor name (as in the name object in this case).

Anyway, if you prefer we can use the filename instead. In which case I agree with your proposed solution.

Reply:
Ah! Ok, ya I think let's stick to the filename rather than descriptor object name, since I agree it'd be nice to not change the descriptor name but I don't want to have spaces in an ID :)

What do you think?

Reply:
I agree. It's true its not ideal at all to have white-spaces in an ID.

So good with me :) I'll make the changes as you stated, using your piece of code.

Reply:
Great!!!